### PR TITLE
Driver improvement: QDac sync

### PR DESCRIPTION
**Pull request by WilliamHPNielsen**
Make sync pulses available

Changes proposed in this pull request:
- Add sync output
- Slightly improve speed of ramping by removing a redundant `get` command

To come:
- [x] notebook documenting changes

@jenshnielsen

### CODE REVIEW

File: qcodes/instrument_drivers/QDev/QDac.py [hunk 1]
```diff
@@ -424,6 +471,13 @@ def _rampvoltage(self, chan, fg, setvoltage, ramptime):
                                             amplitude,
                                             offset)
 
+        if chan in [syn[0] for syn in self._syncoutputs]:
+            syncing = True
+            sync = [syn[1] for syn in self._syncoutputs if syn[0] == chan][0]
+            self.write('syn {} {} 0 {}'.format(sync, fg, 10))
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/QDev/QDac.py]: What is 10 here?

File: qcodes/instrument_drivers/QDev/QDac.py [hunk 2]
```diff
@@ -438,6 +492,9 @@ def _rampvoltage(self, chan, fg, setvoltage, ramptime):
         self.write(funmssg)
         self.parameters['ch{:02}_v'.format(chan)]._save_val(setvoltage)
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/QDev/QDac.py]: What happens if you interrupt during this before sync is reset?

File: qcodes/instrument_drivers/QDev/QDac.py [hunk 3]
```diff
@@ -407,12 +456,10 @@ def _rampvoltage(self, chan, fg, setvoltage, ramptime):
...
         # can jump to its max voltage
-        if ramptime < 0.002:
+        if ramptime <= 0.002:
             ramptime = 0
-
-        # .get is slower than .get_latest, but safe if a ramp is
-        # interrupted
-        v_start = self.parameters['ch{:02}_v'.format(chan)].get()
+            log.warning('Cancelled a ramp with a ramptime of '
+                        '{} s'.format(ramptime))
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/QDev/QDac.py]: Make it more explicit that no voltage change is performed

File: qcodes/instrument_drivers/QDev/QDac.py (hunk 1 above)
Comment (WilliamHPNielsen) [inline qcodes/instrument_drivers/QDev/QDac.py]: 10 ms sync pulse duration.

File: qcodes/instrument_drivers/QDev/QDac.py (hunk 2 above)
Comment (WilliamHPNielsen) [inline qcodes/instrument_drivers/QDev/QDac.py]: Then your sync is still assigned.

File: qcodes/instrument_drivers/QDev/QDac.py (hunk 1 above)
Comment (jenshnielsen) [inline qcodes/instrument_drivers/QDev/QDac.py]: Ok, I suggest making a it variable called sync duration to make this explicit

File: qcodes/instrument_drivers/QDev/QDac.py (hunk 2 above)
Comment (WilliamHPNielsen) [inline qcodes/instrument_drivers/QDev/QDac.py]: We've agreed that this is not an issue, due to syncs being overriden by subsequent calls to `_ramp_voltage`.

**Pull request merged**